### ISSUE TITLE
Remove deprecated field `rolloutStrategy`

### ISSUE THREAD

**Issue by JorTurFer** [kedacore/keda#3596]
### Proposal

In https://github.com/kedacore/keda/pull/3338 we introduced a new section that extends this. 
We need to remove the deprecated `rolloutStrategy`

**Comment by Ritikaa96** [comment]
Hi, @JorTurFer , I'd like to take this up. What I understood is we have to remove `rolloutStartegy` from :

- apis/keda/v1alpha1/scaledjob_types.go
- controllers/keda/scaledjob_controller.go
- config/crd/bases/keda.sh_scaledjobs.yaml

There is no mention of rolloutStrategy anywhere else so we are safe to deprecate it through these.

**Comment by tomkerkhove** [comment]
Thanks! Unfortunately we cannot do this yet as this is a breaking change and we have to wait for KEDA 3.X.

**Comment by tomkerkhove** [comment]
No problem and thanks for looking for contribution items!

**Comment by zroubalik** [comment]
Honestly I think we can drop it :D it's change in crd which is v1alpha1.

**Comment by JorTurFer** [comment]
IMO, we should keep at least 1 version the old code with a warning to avoid breaking changes, giving some time (2-3 months) to update the manifest. Something like, `WARNING: this field XXX has been deprecated and wil be removed in vYYYY`(like k8s does)

**Comment by JorTurFer** [comment]
I mean, I agree with removing the fields, but giving info to the users and time to update, we could improve the user experience. In the case of we notify it and they don't update their manifest, it's totally their fault.
Anyway, we should document this also in https://github.com/kedacore/governance/pull/70

**Comment by zroubalik** [comment]
Absolutely, this is the way.

**Comment by tomkerkhove** [comment]
Which is a good reminder to ask to re-review open comments on https://github.com/kedacore/governance/pull/70 :)

**Comment by JorTurFer** [comment]
Based on our deprecation policies, I guess this should be removed in v2.10

**Comment by tomkerkhove** [comment]
@JorTurFer No, it needs a new API version to remove it as it's part of our CRD AFAIK

**Comment by JorTurFer** [comment]
true
